assets/src/edit-story/app/story/actions/useSavePost.js

```diff
@@ -1,14 +1,44 @@
 /**
  * WordPress dependencies
  */
-import { useCallback } from '@wordpress/element';
+import { useCallback, renderToString } from '@wordpress/element';
 import { addQueryArgs } from '@wordpress/url';
 
 /**
  * Internal dependencies
  */
 import { useAPI } from '../../api';
 
+const getStoryMarkupFromPages = ( pages ) => {
+	// @todo look into: We should rather use renderToStaticMarkup here, however, it doesn't seem to be exposed via @wordpress/element.
+	const markup = pages.map( ( page ) => {
+		const { id } = page;
+		return renderToString(
+			<amp-story-page id={ id }>
+				<amp-story-grid-layer template="vertical">
+					{ page.elements.map( ( { type, src, width, height, content } ) => {
+						const style = {
+							position: 'absolute',
+						};
+						// @todo this should be redone by using dynamic tag.
+						if ( 'image' === type ) {
+							return <amp-img key={ 'el' + id } style={ { ...style } } src={ src } layout="fixed" width={ width } height={ height } />;
```

Comment:
These different elements should be different component files and use a switch statement.